### PR TITLE
line charts now span y-axis from min to max value

### PR DESCRIPTION
**Pull request by alexmat**
This is a draft for enhancement request #840 

I chose a 3% margin after trying a few values. This seemed to me to use up the available space without feeling cramped.

### CODE REVIEW

File: fava/static/javascript/charts.js [hunk 1]
```diff
@@ -539,9 +539,14 @@ class LineChart extends BaseChart {
       min(this.data, s => s.values[0].date),
       max(this.data, s => s.values[s.values.length - 1].date),
     ]);
+
+    // Span y-axis min to max value plus margin as percent of either value
+    this.yMarginPercent = 0.03;
```
Comment (yagebu) [inline fava/static/javascript/charts.js]: Don't think we need a variable for this, just use it inline

File: fava/static/javascript/charts.js [hunk 2]
```diff
@@ -539,9 +539,14 @@ class LineChart extends BaseChart {
...
+    this.minDataValue = min(this.data, d => min(d.values, x => x.value));
```
Comment (yagebu) [inline fava/static/javascript/charts.js]:
```suggestion
    const minDataValue = min(this.data, d => min(d.values, x => x.value));
```

File: fava/static/javascript/charts.js [hunk 3]
```diff
@@ -539,9 +539,14 @@ class LineChart extends BaseChart {
...
+    this.maxDataValue = max(this.data, d => max(d.values, x => x.value));
     this.y.domain([
-      Math.min(0, min(this.data, d => min(d.values, x => x.value))),
-      Math.max(0, max(this.data, d => max(d.values, x => x.value))),
+      this.minDataValue - (this.minDataValue * this.yMarginPercent),
+      this.maxDataValue + (this.maxDataValue * this.yMarginPercent)
```
Comment (yagebu) [inline fava/static/javascript/charts.js]: Our linter complains, you should add a comma at the end of the line

File: fava/static/javascript/charts.js [hunk 4]
```diff
@@ -539,9 +539,14 @@ class LineChart extends BaseChart {
...
     ]);
+
+    // Span y-axis min to max value plus margin as percent of either value
+    this.yMarginPercent = 0.03;
+    this.minDataValue = min(this.data, d => min(d.values, x => x.value));
+    this.maxDataValue = max(this.data, d => max(d.values, x => x.value));
     this.y.domain([
-      Math.min(0, min(this.data, d => min(d.values, x => x.value))),
-      Math.max(0, max(this.data, d => max(d.values, x => x.value))),
+      this.minDataValue - (this.minDataValue * this.yMarginPercent),
```
Comment (yagebu) [inline fava/static/javascript/charts.js]: I think the margin should scale with the difference of min and max, not just the min and max. Let's say the min is 0, this wouldn't do anything.

**Comment by yagebu** [comment]
Thanks! I've added some comments

**Comment by alexmat** [comment]
Thanks for the great feedback! I applied the changes you mentioned but I still used a yMargin constant. My argument in favor of it is that the value is used more than once and involves a "magic value". I am happy to inline it if you still feel it should go that route.

File: fava/static/javascript/charts.js (hunk 1 above)
Comment (alexmat) [reply fava/static/javascript/charts.js]: My argument in favor of it is that the value is used more than once and involves a "magic value". I am happy to inline it if you still feel it should go that route.

File: fava/static/javascript/charts.js (hunk 3 above)
Comment (alexmat) [reply fava/static/javascript/charts.js]: How do I run the linter so I can check before commiting?

File: fava/static/javascript/charts.js (hunk 4 above)
Comment (alexmat) [reply fava/static/javascript/charts.js]: Good catch. Fixed.

File: fava/static/javascript/charts.js (hunk 2 above)
Comment (alexmat) [reply fava/static/javascript/charts.js]: Makes sense. Fixed.

File: fava/static/javascript/charts.js (hunk 1 above)
Comment (alexmat) [reply fava/static/javascript/charts.js]: It also saves an extra calculation haha

File: fava/static/javascript/charts.js (hunk 3 above)
Comment (aumayr) [reply fava/static/javascript/charts.js]: In the root folder of the project: `make lint`

Comment (alexmat) [reply fava/static/javascript/charts.js]:
```
  lint: commands succeeded
  congratulations :)
```
 Looks like it's happy. Thanks.

File: fava/static/javascript/charts.js (hunk 1 above)
Comment (alexmat) [reply fava/static/javascript/charts.js]: After playing with the values a bit, 5% looks better. Code also looks nicer with the value inline. I'd advocate for the variable more if it was used somewhere outside the local scope, as it is, it seems clear enough.

**Comment by yagebu** [comment]
Thanks again :)

**Pull request merged**